src/accelerate/inference.py

```diff
@@ -59,9 +62,42 @@ def build_pipeline(model, split_points, args, kwargs) -> PipelineStage:
 def pippy_forward(forward, *args, **kwargs):
     state = PartialState()
     output = None
+
+    def _find_batch_size(arg):
+        try:
+            return find_batch_size(arg)
+        except (ValueError, TypeError):
+            pass
+        return None
+
+    def _pad_inputs(args):
+        # Slice and copy the last input
+        extra = slice_tensors(
+            args,
+            slice(state.num_processes, state.num_processes + 1),
+        )
+        # Concat it to `args`
+        args = concatenate([args, extra])
```

Comment:
You are only concatenating one extra input. it should be `state.num_processes - (found_batch_size % state.num_processes` no ?